### PR TITLE
fix: inline variables are wrongly transformed as ctxVars

### LINKED ISSUE #3258: Can't define variables in inline event handler: "Property does not exist on type"
When defining a variable like this:
```vue
@click="() => {
  const x = 1
  if (x === 1) {
  }
}"
```
it yields two errors:
* `'x' is declared but its value is never read. ts(6133)`
* `Property 'x' does not exist on type 'MyComponent'. ts(2339)`

The code using such variable **works correctly**, and I've been using this syntax since Vue 2 with **Vetur**. After migrating to Vue 3 with **Volar**, this type safety and inference in VS Code fails.

image

### PR DESCRIPTION
**Pull request by so1ve**
closes #3258

### CODE REVIEW

File: packages/vue-language-core/src/utils/transform.ts [hunk 1]
```diff
@@ -39,7 +39,10 @@ export function walkInterpolationFragment(
 	};
 	ast.forEachChild(node => walkIdentifiers(ts, node, varCb, localVars));
 
-	ctxVars = ctxVars.sort((a, b) => a.offset - b.offset);
+	// Avoid duplicate ctx vars
+	// Fixes https://github.com/vuejs/language-tools/issues/3258
+	const localVarsKeys = Object.keys(localVars);
+	ctxVars = ctxVars.filter(v => !localVarsKeys.includes(v.text)).sort((a, b) => a.offset - b.offset);
```
Comment (rchl) [inline packages/vue-language-core/src/utils/transform.ts:45]: Wouldn't it make sense to avoid pushing to `ctxVars` in the first place above?

Comment (so1ve) [reply packages/vue-language-core/src/utils/transform.ts:45]: Makes sense. Update later

Comment (so1ve) [reply packages/vue-language-core/src/utils/transform.ts]: Done

**Comment by so1ve** [comment]
Oops. Why?

**Comment by johnsoncodehk** [comment]
If `localVars[id.text]` is 0 or undefined, it means that the current code scope cannot access the local variable, so this is a problem that `walkIdentifiers` does not collect variables correctly.

**Comment by so1ve** [comment]
Thanks for your explanation! Now I know how auto unwrapping works :)

**Pull request merged**